### PR TITLE
Rename package to @sinonjs/fake-clock

### PR DESCRIPTION
**Pull request by mroderick**
This PR renames the package to `@sinonjs/fake-clock`.

#### Purpose

It was suggested that we rename this package to have a less confusing name (I can't recall who suggested it, or in which communication channel).

I will soon be doing a larger effort in improving the documentation of Sinon and associated libraries. I thought it would be beneficial to that effort to get the rename of packages out of the way first.

See https://github.com/sinonjs/sinon/issues/1651

#### Plan

1. Rename repository
1. Merge this branch
1. Bump the major version
1. Publish the renamed package to `npm` registry
1. Deprecate the `lolex` package in the `npm` registry
1. Update `sinon` to use the new package

#### How to verify

1. Check out this branch
1. (this repo): `npm ci`
1. (this repo): `npm link`
1. (sinon repo): `npm link @sinonjs/fake-clock`
1. (sinon repo): Update the `lib/sinon/util/fake-timers.js` to use `@sinonjs/fake-clock` instead of `lolex`
1. (sinon repo): `npm test`
1. Observe tests pass

### CODE REVIEW

**Comment by mantoni** [comment]
I guess we‘ll rename `nise` to `@sinonjs/fake-xhr`?

**Comment by mroderick** [comment]
> I guess we‘ll rename `nise` to `@sinonjs/fake-xhr`?

https://github.com/sinonjs/nise/pull/131 ... let's discuss that package in that PR

File: README.md [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
 
-In addition in browser environment lolex provides a `performance` implementation that gets its time from the clock. In Node environments lolex provides a `nextTick` implementation that is synchronized with the clock - and a `process.hrtime` shim that works with the clock.
+In addition in browser environment `@sinonjs/fake-clock` provides a `performance` implementation that gets its time from the clock. In Node environments FakeClock provides a `nextTick` implementation that is synchronized with the clock - and a `process.hrtime` shim that works with the clock.
 
-Lolex can be used to simulate passing time in automated tests and other
+`@sinonjs/fake-clock` can be used to simulate passing time in automated tests and other
 situations where you want the scheduling semantics, but don't want to actually
-wait (however, from version 2.0 lolex supports those of you who would like to wait too).
+wait (however, from version 2.0 `@sinonjs/fake-clock` supports those of you who would like to wait too).
```
Comment (SimenB) [inline README.md]: this is just weird, isn't it? There won't be a version 2 of this new package

File: fake-clock.js [hunk 2]
```diff
@@ -1888,9 +1888,9 @@ function withGlobal(_global) {
             typeof config === "number"
         ) {
             throw new TypeError(
```
Comment (SimenB) [inline fake-clock.js]: just ditch this whole thing?

File: README.md (hunk 1 above)
Comment (mroderick) [reply README.md]: Yeah. Perhaps it should just be dropped

File: fake-clock.js (hunk 2 above)
Comment (mroderick) [reply fake-clock.js]: People have probably moved on by now, so that is most likely safe to do :)

**Comment by mantoni** [comment]
Sorry to come up with this so late, after the work has already been done, but I'd like to discuss if `@sinonjs/fake-timers` would be an even better name since it would match `sinon.useFakeTimers()`. What do you think?

**Comment by mroderick** [comment]
> Sorry to come up with this so late, after the work has already been done, but I'd like to discuss if `@sinonjs/fake-timers` would be an even better name since it would match `sinon.useFakeTimers()`. What do you think?

That also works. I don't really have a preference. What says @SimenB and @fatso83?

**Comment by mroderick** [comment]
This fails in `master` for me: `npm run test-cloud`.

Until I have a chance to fix that, I am not going to publish a release under the new name.

**Comment by mantoni** [comment]
Aaah, our favorite browser screaming for attention again 😟 👶

**Comment by mroderick** [comment]
This has been published to `npm` as `@sinonjs/fake-timers@6.0.0`.

@SimenB let us know how it goes with using the package in the Jest timers package?

**Comment by SimenB** [comment]
The only "breaking" change was the rename, right? If yes, happy to update 🙂

**Comment by mroderick** [comment]
I've published the last commit 6a1248cf02867d76cf596b299ed74841eceff236 before the rename as `lolex@6.0.0`.

I have deprecated with the following command:

```
npm deprecate lolex@6.0.0 "lolex has been renamed to @sinonjs/fake-timers. No API changes made. Please use the new package instead"
```

**Comment by mroderick** [comment]
> The only "breaking" change was the rename, right? If yes, happy to update

Yep. Only a rename. No other changes made.

**Comment by SimenB** [comment]
perfect! 🙂

**Comment by mroderick** [comment]
> EDIT: Actually, we'll be stuck until a version of the types is published to from DefinitelyTyped... Should we just ship with typings? #220

Yes, and no. I am doing some work this month that hopefully will result in:

1. Better API documentation for the projects
1. Better support in IDEs like VSCode / IntelliJ
1. Enabling automatic generation of the TypeScript artefacts, instead of that being a manual process

It's part of a longer effort, more to come.

(If you put an oar in this water, be prepared to **pull** on it).

**Comment by fatso83** [comment]
See #365

**Pull request merged**